src/category_theory/sites/types.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import category_theory.sites.sheaf
+
+/-!
+# Grothendieck Topology and Sheaves on the Category of Types
+
+In this file we define the grothendieck topology on the category of types,
+and construct the canonical function that sends a type to a sheaf over
+the category of types.
+-/
+
+universe u
+
+namespace category_theory
+
+/-- The Grothendieck topology associated to the category of all types. -/
+def types_grothendieck_topology : grothendieck_topology (Type u) :=
+{ sieves := λ α S, ∀ x : α, S (λ _ : punit, x),
+  top_mem' := λ α x, trivial,
+  pullback_stable' := λ α β S f hs x, hs (f x),
+  transitive' := λ α S hs R hr x, hr (hs x) punit.star }
+
+/-- The discrete sieve on a type. -/
+def discrete_sieve (α : Type u) : sieve α :=
+{ arrows := λ β f, ∃ x, ∀ y, f y = x,
+  downward_closed' := λ β γ f ⟨x, hx⟩ g, ⟨x, λ y, hx $ g y⟩ }
+
+lemma discrete_sieve_mem (α : Type u) : discrete_sieve α ∈ types_grothendieck_topology α :=
+λ x, ⟨x, λ y, rfl⟩
+
+/-- The discrete presieve on a type. -/
+def discrete_presieve (α : Type u) : presieve α :=
+λ β f, ∃ x : β, ∀ y : β, y = x
+
+lemma generate_discrete_presieve_mem (α : Type u) :
+  sieve.generate (discrete_presieve α) ∈ types_grothendieck_topology α :=
+λ x, ⟨punit, id, λ _, x, ⟨punit.star, λ _, subsingleton.elim _ _⟩, rfl⟩
+
+namespace presieve
+
+theorem is_sheaf_yoneda' {α : Type u} : is_sheaf types_grothendieck_topology (yoneda.obj α) :=
```

Comment:
You could import `canonical.lean` and show that this topology is subcanonical if you'd like as well.

Reply:
Done.